hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -699,7 +699,14 @@ protected void runAnyPendingClustering(HoodieTable table) {
   protected void writeTableMetadata(HoodieTable table, String instantTime, String actionType, HoodieCommitMetadata metadata, HoodieData<WriteStatus> writeStatuses) {
     checkArgument(table.isTableServiceAction(actionType, instantTime), String.format("Unsupported action: %s.%s is not table service.", actionType, instantTime));
     context.setJobStatus(this.getClass().getSimpleName(), "Committing to metadata table: " + config.getTableName());
-    table.getMetadataWriter(instantTime).ifPresent(w -> ((HoodieTableMetadataWriter) w).update(metadata, writeStatuses, instantTime));
+    Option<HoodieTableMetadataWriter> metadataWriterOpt = table.getMetadataWriter(instantTime);
+    if (metadataWriterOpt.isPresent()) {
+      try (HoodieTableMetadataWriter metadataWriter = metadataWriterOpt.get()) {
+        metadataWriter.update(metadata, writeStatuses, instantTime);
+      } catch (Exception e) {
```

Comment:
Should we let this throw the exception instead of catching it, to match the earlier behavior?

Reply:
we are throwing HoodieException right. should suffice I guess. atleast we did not want to swallow it.

Reply:
If the exception thrown by that method is a specific impl of HoodieException, we lose that context so wanted to call that out

Reply:
nvm. will fix it to be same as before

Reply:
due to auto closeable, we have to declare catch or have to explicitly throw